src/common/user-store.ts

```diff
@@ -175,15 +208,59 @@ export class UserStore extends BaseStore<UserStoreModel> {
       this.kubeConfigPath = kubeConfigPath;
     }
     this.seenContexts.replace(seenContexts);
-    Object.assign(this.preferences, preferences);
+
+    if (!preferences) {
+      return;
+    }
+
+    this.httpsProxy = preferences.httpsProxy;
+    this.shell = preferences.shell;
+    this.colorTheme = preferences.colorTheme;
+    this.localeTimezone = preferences.localeTimezone;
+    this.allowUntrustedCAs = preferences.allowUntrustedCAs;
+    this.allowTelemetry = preferences.allowTelemetry;
+    this.downloadMirror = preferences.downloadMirror;
+    this.downloadKubectlBinaries = preferences.downloadKubectlBinaries;
+    this.downloadBinariesPath = preferences.downloadBinariesPath;
+    this.kubectlBinariesPath = preferences.kubectlBinariesPath;
+    this.openAtLogin = preferences.openAtLogin;
+
+    if (preferences.hiddenTableColumns) {
+      for (const [tableId, columnIds] of preferences.hiddenTableColumns) {
+        if (this.hiddenTableColumns.has(tableId)) {
+          this.hiddenTableColumns.get(tableId).replace(columnIds);
+        } else {
+          this.hiddenTableColumns.set(tableId, new ObservableToggleSet(columnIds));
+        }
+      }
+    }
   }
 
   toJSON(): UserStoreModel {
+    const hiddenTableColumns: [string, string[]][] = [];
+
+    for (const [key, values] of this.hiddenTableColumns.entries()) {
+      hiddenTableColumns.push([key, Array.from(values)]);
+    }
+
     const model: UserStoreModel = {
       kubeConfigPath: this.kubeConfigPath,
       lastSeenAppVersion: this.lastSeenAppVersion,
       seenContexts: Array.from(this.seenContexts),
-      preferences: this.preferences,
+      preferences: {
+        httpsProxy: toJS(this.httpsProxy),
+        shell: toJS(this.shell),
+        colorTheme: toJS(this.colorTheme),
+        localeTimezone: toJS(this.localeTimezone),
+        allowUntrustedCAs: toJS(this.allowUntrustedCAs),
+        allowTelemetry: toJS(this.allowTelemetry),
+        downloadMirror: toJS(this.downloadMirror),
+        downloadKubectlBinaries: toJS(this.downloadKubectlBinaries),
+        downloadBinariesPath: toJS(this.downloadBinariesPath),
+        kubectlBinariesPath: toJS(this.kubectlBinariesPath),
+        openAtLogin: toJS(this.openAtLogin),
```

Comment:
Why there  are `toJS` calls? Below `toJS(model)` should do the job.

Reply:
Oh I missed it bellow, will remove